examples/vision/cutmix.py

```diff
@@ -0,0 +1,383 @@
+    if target_w == 0:
+        target_w += 1
+
+    return bbx1, bby1, target_h, target_w
+
+
+@tf.function
+def cutmix(a, b):
+
+    (image1, label1), (image2, label2) = a, b
+
+    alpha = [1.0]
+    beta = [1.0]
+
+    ## Get sample from beta distribution
+    dist = tfd.Beta(alpha, beta)
+    ## Lambda
+    l = dist.sample(1)[0][0]
+
+    ## Get bbox ofsets and heights and widths
+    bbx1, bby1, target_h, target_w = get_box(l)
+
+    ## Get patch from image2
+    crop2 = tf.image.crop_to_bounding_box(image2, bby1, bbx1, target_h, target_w)
+    ## Pad the patch with same offset
+    image2 = tf.image.pad_to_bounding_box(crop2, bby1, bbx1, IMG_SHAPE, IMG_SHAPE)
+    ## Get patch from image1
+    crop1 = tf.image.crop_to_bounding_box(image1, bby1, bbx1, target_h, target_w)
+    ## Pad the patch with same offset
+    img1 = tf.image.pad_to_bounding_box(crop1, bby1, bbx1, IMG_SHAPE, IMG_SHAPE)
+
+    ## Subtract the patch from image1 so that patch from image2 can be put on instead
+    image1 = image1 - img1
+    ## Add modified image1 and image2 to get cutmix image
+    image = image1 + image2
+
+    ## Adjust lambda according to pixel ration
+    l = 1 - (target_w * target_h) / (IMG_SHAPE * IMG_SHAPE)
+    l = tf.cast(l, tf.float32)
+
+    ## Combine labels
+    label = l * label1 + (1 - l) * label2
+
+    return image, label
+
+
+"""
+**Note** that here , we are combining two images to create a single one.
+"""
+
+"""
+## Visualize the new dataset after applying CutMix Augmentation.
+"""
+
+# Create the new dataset using our `mix_up` utility
+train_ds_cmu = (
+    train_ds.shuffle(1024)
+    .map(cutmix, num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+# Let's preview 9 samples from the dataset
+image_batch, label_batch = next(iter(train_ds_cmu))
+plt.figure(figsize=(10, 10))
+for i in range(9):
+    ax = plt.subplot(3, 3, i + 1)
+    plt.title(class_names[np.argmax(label_batch[i])])
+    plt.imshow(image_batch[i])
+    plt.axis("off")
+
+"""
+## Define the Model
+"""
+
+
+def resnet_layer(
+    inputs,
+    num_filters=16,
+    kernel_size=3,
+    strides=1,
+    activation="relu",
+    batch_normalization=True,
+    conv_first=True,
+):
+
+    conv = keras.layers.Conv2D(
+        num_filters,
+        kernel_size=kernel_size,
+        strides=strides,
+        padding="same",
+        kernel_initializer="he_normal",
+        kernel_regularizer=keras.regularizers.l2(1e-4),
+    )
+
+    x = inputs
+    if conv_first:
+        x = conv(x)
+        if batch_normalization:
+            x = keras.layers.BatchNormalization()(x)
+        if activation is not None:
+            x = keras.layers.Activation(activation)(x)
+    else:
+        if batch_normalization:
+            x = keras.layers.BatchNormalization()(x)
+        if activation is not None:
+            x = keras.layers.Activation(activation)(x)
+        x = conv(x)
+    return x
+
+
+def resnet_v20(input_shape, depth, num_classes=10):
```

Comment:
Why not just use a Keras Applications model, to simplify this example? The main point of the example is the data augmentation, not the model itself.

Reply:
I had tried with ResNet50. Implementation of ResNet20 is specifically configured for the CIFAR10 dataset. With ResNet50 it is leading to representation collapse. So I better followed the general setting.

Reply:
Have you tried a smaller model, like MobileNet?